apps/web/src/scenes/NftDetailPage/NftDetailAsset.tsx

```diff
@@ -201,33 +210,73 @@ function NftDetailAsset({ tokenRef, hasExtraPaddingForNote }: Props) {
           }
         }
         media @required(action: THROW) {
+          ... on Media {
+            dimensions {
+              width
+              height
+            }
+          }
           ... on HtmlMedia {
             __typename
           }
         }
 
+        ...useGetPreviewImagesSingleFragment
         ...NftDetailAssetComponentFragment
       }
     `,
     collectionToken.token
   );
 
   const breakpoint = useBreakpoint();
-  const { aspectRatioType } = useContentState();
 
   const contractAddress = token.contract?.contractAddress?.address ?? '';
   const backgroundColorOverride = getBackgroundColorOverrideForContract(contractAddress);
 
   // We do not want to enforce square aspect ratio for iframes https://github.com/gallery-so/gallery/pull/536
   const isIframe = token.media.__typename === 'HtmlMedia';
   const shouldEnforceSquareAspectRatio =
-    !isIframe &&
-    (aspectRatioType !== 'wide' || breakpoint === size.desktop || breakpoint === size.tablet);
+    !isIframe && (breakpoint === size.desktop || breakpoint === size.tablet);
 
   const { handleNftLoaded } = useNftRetry({
     tokenId: token.dbid,
   });
 
+  const imageUrl = useGetSinglePreviewImage({ tokenRef: token, size: 'large' }) ?? '';
+
+  const { cacheLoadedImageUrls, cachedUrls } = useNftPreviewFallbackState();
+  const tokenId = token.dbid;
+
+  const hasPreviewUrl = cachedUrls[tokenId]?.type === 'preview';
+  const hasRawUrl = cachedUrls[tokenId]?.type === 'raw';
+  const shouldShowShimmer = !(hasPreviewUrl || hasRawUrl);
+
+  const resultDimensions = useMemo(() => {
+    const serverSourcedDimensions = token.media?.dimensions;
+    if (serverSourcedDimensions?.width && serverSourcedDimensions.height) {
+      return fitDimensionsToContainerContain({
+        container: { width: DESKTOP_TOKEN_SIZE, height: DESKTOP_TOKEN_SIZE },
+        source: {
+          width: serverSourcedDimensions.width,
+          height: serverSourcedDimensions.height,
+        },
+      });
+    }
+
+    return {
+      height: DESKTOP_TOKEN_SIZE,
+      width: DESKTOP_TOKEN_SIZE,
+    };
+  }, [token.media?.dimensions]);
```

Comment:
if this ends up getting more complicated to incorporate mobile dimensions, let's turn it into its own reusable hook so the other file can use it too

Reply:
it became a bit more complicated (1 line more), but I don't think making a reusable hook is the best because we can't pass `serverSourcedDimensions` directly into the hook since we'd need to do the if check (on L256) before anyway and create a new object that's a copy of it (as source), so there's no real part that's reusable I think.